test/test_interpreter.cpp

```diff
@@ -120,6 +124,62 @@ TEST_SUITE("inspect_request")
     }
 }
 
+TEST_SUITE("kernel_info_request")
+{
+    TEST_CASE("good_status")
+    {
+        std::vector<const char*> Args = {/*"-v", "resource-dir", "....."*/};
+        xcpp::interpreter interpreter((int)Args.size(), Args.data());
+
+        nl::json result = interpreter.kernel_info_request();
+
+        REQUIRE(result["implementation"] == "xeus-cpp");
+        REQUIRE(result["language_info"]["name"] == "C++");
+        REQUIRE(result["language_info"]["mimetype"] == "text/x-c++src");
+        REQUIRE(result["language_info"]["codemirror_mode"] == "text/x-c++src");
+        REQUIRE(result["language_info"]["file_extension"] == ".cpp");
+        REQUIRE(result["status"] == "ok");
+    }
+
+}
+
+TEST_SUITE("shutdown_request")
+{
+    TEST_CASE("good_status")
+    {
+        std::vector<const char*> Args = {/*"-v", "resource-dir", "....."*/};
+        xcpp::interpreter interpreter((int)Args.size(), Args.data());
+
+        REQUIRE_NOTHROW(interpreter.shutdown_request());
+    }
+
+}
+
+TEST_SUITE("is_complete_request")
+{
+    TEST_CASE("incomplete_code")
+    {
+        std::vector<const char*> Args = {/*"-v", "resource-dir", "....."*/};
+        xcpp::interpreter interpreter((int)Args.size(), Args.data());
+
+        std::string code = "int main() \\";
+        nl::json result = interpreter.is_complete_request(code);
+        REQUIRE(result["status"] == "incomplete");
+
+                std::cerr << "qweqw"<< std::endl;
+    }
+
+    TEST_CASE("complete_code")
+    {
+        std::vector<const char*> Args = {/*"-v", "resource-dir", "....."*/};
+        xcpp::interpreter interpreter((int)Args.size(), Args.data());
+
+        std::string code = "int main() {}";
+        nl::json result = interpreter.is_complete_request(code);
```

Comment:
Do you intend to test the `complete_request` or the `is_complete_request` here

Reply:
`is_complete_request`

Reply:
Works.